### ISSUE TITLE
content preview input field doesn't allow relative urls

### ISSUE THREAD

**Issue by raffaelj** [agentejo/cockpit#1311]
I want to use a relative url `/livePreview` as content preview url and the collection settings page complains about it because it has an `input type="url"`. If I change it to `type="text"` in my browser or if I change the `pages.collection.php` manually, my content preview works as expected.

But now I'm not able to save changes via admin ui because the preview input field complains in a not selected/not visible tab without a visible error message. If I change it back to `type="text"`, I can work with it.

The reason for using a relative url is to keep my local dev setup in sync with the production server without having to change `http://localhost/livePreview` to `https://example.com/livePreview` all the time.

**Comment by raffaelj** [comment]
@artur Thanks for the fast [update][1], but that doesn' work for me. A quick check in my console gave me:

```js
SITE_URL
"http://localhost/data/cp"
App.base('/')
"/cockpit/"
```

So `site://` and `base://` won't work. What I need is `http://localhost` or an empty string.

My setup:

My frontend CpMultiplane is in the root. Cockpit as the backend is in a subfolder.

```text
cockpit/
    defines.php --> COCKPIT_ENV_ROOT => '../data/cp'
    ...
data/
    cp/
        addons/
        config/
        storage/
    mp/
        ...
defines.php --> MP_ENV_ROOT => './data/mp'
index.php
...
```

The frontend provides an endpoint `/livePreview`, which is available under `http://localhost/livePreview`.

And as I wrote, it works if I change it manually, but then the collection settings page becomes unusable because the form won't submit with an invalid url.

[1]: https://github.com/agentejo/cockpit/commit/e7b5cdca63c6ebae17dc2bbfa082417624de2ca8

**Comment by pauloamgomes** [comment]
@raffaelj not sure if that helps, but in the past I had a similar issue, I solved that with an override mechanism for content preview urls - https://github.com/pauloamgomes/CockpitCMS-Helpers#content-preview-url-override

**Comment by raffaelj** [comment]
@pauloamgomes Thanks for the hint, but I want to avoid overwriting javascript with partials. I used too much hacks and it gets confusing over time.

Also I found a way with the new mechanic: `base://../livePreview?token=xx`, which translates to `/cockpit/../livePreview?token=xx`.

My problem is solved, but it feels a bit counterintuitive. So I leave the issue open. I still think, that pure, relative urls should be allowed in a browser-validated url input field.
